evm/p2p/protocol.py

```diff
@@ -85,27 +95,28 @@ class Protocol:
     logger = logging.getLogger("evm.p2p.protocol.Protocol")
     name = None  # type: bytes
     version = None  # type: int
+    cmd_length = None  # type: int
     handshake_msg_type = None  # type: Type[Command]
     # List of Command classes that this protocol supports.
     _commands = []  # type: List[Type[Command]]
 
-    def __init__(self, peer, cmd_id_offset):
+    def __init__(self, peer: 'BasePeer', cmd_id_offset: int) -> None:
```

Comment:
Is `'BasePeer'` a string due to circular imports?

Reply:
Yep